From 367424c09bb0a5992458669d349cadc07b35a8aa Mon Sep 17 00:00:00 2001
From: Neal Liu <neal_liu@aspeedtech.com>
Date: Tue, 11 Jan 2022 15:10:05 +0800
Subject: [PATCH] Fix HID gadget device remote wakeup issue

Enable Bit(6): Self-powered & Bit(5): Remote Wakeup
to allow host users to decide whether HID gadget device
can wakeup host or not.
Host Controller would set REMOTE_WAKEUP feature before
suspend, and the devices acked accordingly.

Tested: Host can be waked up by HID device when KVM client
is connected.

Signed-off-by: Neal Liu <neal_liu@aspeedtech.com>
Change-Id: Iddbc456a970226ebd1f419cc6ee4ac7dc8ab989f
---
 create_usbhid.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/create_usbhid.sh b/create_usbhid.sh
index d1fa4e0..d1b9841 100644
--- a/create_usbhid.sh
+++ b/create_usbhid.sh
@@ -107,7 +107,7 @@ create_hid() {
     mkdir configs/c.1
     mkdir configs/c.1/strings/0x409
 
-    echo 0x80 > configs/c.1/bmAttributes
+    echo 0xe0 > configs/c.1/bmAttributes
     echo 200 > configs/c.1/MaxPower
     echo "" > configs/c.1/strings/0x409/configuration
 
-- 
2.25.1

